lib/semian/pid_controller.rb

```diff
@@ -143,7 +139,8 @@ def calculate_p_value(current_error_rate)
         # P = (error_rate - ideal_error_rate) - rejection_rate
         # P increases when: error_rate > ideal
         # P decreases when: rejection_rate > 0 (feedback mechanism)
-        (current_error_rate - ideal_error_rate) - @rejection_rate
+        delta_error = current_error_rate - ideal_error_rate
+        delta_error - (1 - delta_error) * @rejection_rate
```

Comment:
This is the key change: an elastic defensiveness that stretches and expands based on how bad the error rate is